beacon-chain/blockchain/process_block.go

```diff
@@ -126,11 +114,23 @@ func (s *Service) postBlockProcess(ctx context.Context, signed interfaces.ReadOn
 	})
 
 	defer reportAttestationInclusion(b)
-	//only handle epoch boundary if the incoming block is canonical,
-	//otherwise this will be handled by lateBlockTasks.
 	if headRoot == blockRoot {
-		if err := s.handleEpochBoundary(ctx, postState.Slot(), postState, blockRoot[:]); err != nil {
-			return errors.Wrap(err, "could not handle epoch boundary")
+		// Updating next slot state cache can happen in the background
+		// except in the epoch boundary in which case we lock to handle
+		// the shuffling and proposer caches updates.
+		// We handle these caches only on canonical
+		// blocks, otherwise this will be handled by lateBlockTasks
+		slot := postState.Slot()
+		if (slot+1)%params.BeaconConfig().SlotsPerEpoch != 0 {
+			go updateNextSlotCacheOnBlock(blockRoot, postState)
```

Comment:
I think it's cleaner not having the helper function and do
```go
go func() {
			slotCtx, cancel := context.WithTimeout(context.Background(), slotDeadline)
			defer cancel()
			if err := transition.UpdateNextSlotCache(slotCtx, blockRoot[:], postState); err != nil {
				log.WithError(err).Debug("could not update next slot state cache")
			}
		}()
```

Reply:
+1 , that is cleaner

Reply:
Also there should be a slot helper for this:
`if (slot+1)%params.BeaconConfig().SlotsPerEpoch != 0` , would be nicer to use that

Reply:
I'll use `!slots.IsEpochStart(slot+1)`